### PR TITLE
Code skeleton + config setup

### PR DESCRIPTION
**Pull request by chongr**
discussed in #4 wasn't sure exactly how to seperate out the config values so just wrote the ones I was thinking of so we could discuss where they should go.

Also provide skeleton for the repo so work can be split and added to the different interfaces

### CODE REVIEW

File: gcp/gcp.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	MetadataBucket string
+
+	// bucket where follower put intermediate files to be moved by leader
+	StagingBucket string
+
+	// final bucket for firehose files
+	OutputBucket string
+
+	// interval when leader moves files to final bucket
+	ScrapeInterval int
```
Comment (chongr) [inline gcp/gcp.go]: im thinking scrapeInterval and FlushAfter have nothing to do with the cloud backend  and can be moved out but wasn't exactly sure what you were thinking for those @maroux

File: main.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+import (
+	"fmt"
+)
+
+// FirehoseBackend is used for consuming messages from a transport and read/write to storage
+type FirehoseBackend interface {
+}
+
+type FirehoseProcess struct {
```
Comment (maroux) [inline main.go]: not sure about difference between `FirehoseProcess` and `Firehose` structs.

File: gcp/gcp.go (hunk 1 above)
Comment (maroux) [reply gcp/gcp.go]: Yeah sounds like these should be moved out into the higher level of firehose config perhaps.

File: main.go (hunk 2 above)
Comment (chongr) [reply main.go]: I was sort of modeling it off this https://github.com/cloudchacho/hedwig-go/blob/main/consumer.go#L24, guessing process will have more in it later on?  I can probably just make it one too

Comment (chongr) [reply main.go]: changed to one

File: gcp/gcp.go (hunk 1 above)
Comment (chongr) [reply gcp/gcp.go]: moved to higher lvl

**Comment by chongr** [comment]
This diff does not deal with some discussion in #4 about how to actually load these vars - that can be added in another diff this just provides where those vars will go (I'll add another task for that)

**Review by maroux — APPROVED** [review]

**Pull request merged**